diff --git a/llamacpp/native/src/server/server-http.cpp b/llamacpp/native/src/server/server-http.cpp
index 62250571..77e54d19 100644
--- a/llamacpp/native/src/server/server-http.cpp
+++ b/llamacpp/native/src/server/server-http.cpp
@@ -8,10 +8,6 @@
 #include <string>
 #include <thread>
 
-// auto generated files (see README.md for details)
-#include "index.html.gz.hpp"
-#include "loading.html.hpp"
-
 //
 // HTTP implementation using cpp-httplib
 //
@@ -175,26 +171,17 @@ bool server_http_context::init(const common_params & params) {
     auto middleware_server_state = [this](const httplib::Request & req, httplib::Response & res) {
         bool ready = is_ready.load();
         if (!ready) {
-            auto tmp = string_split<std::string>(req.path, '.');
-            if (req.path == "/" || tmp.back() == "html") {
-                res.set_content(reinterpret_cast<const char*>(loading_html), loading_html_len, "text/html; charset=utf-8");
-                res.status = 503;
-            } else if (req.path == "/models" || req.path == "/v1/models" || req.path == "/api/tags") {
-                // allow the models endpoint to be accessed during loading
-                return true;
-            } else {
-                res.status = 503;
-                res.set_content(
-                    safe_json_to_str(json {
-                        {"error", {
-                            {"message", "Loading model"},
-                            {"type", "unavailable_error"},
-                            {"code", 503}
-                        }}
-                    }),
-                    "application/json; charset=utf-8"
-                );
-            }
+            res.status = 503;
+            res.set_content(
+                safe_json_to_str(json {
+                    {"error", {
+                        {"message", "Loading model"},
+                        {"type", "unavailable_error"},
+                        {"code", 503}
+                    }}
+                }),
+                "application/json; charset=utf-8"
+            );
             return false;
         }
         return true;
@@ -253,7 +240,6 @@ bool server_http_context::init(const common_params & params) {
                     // COEP and COOP headers, required by pyodide (python interpreter)
                     res.set_header("Cross-Origin-Embedder-Policy", "require-corp");
                     res.set_header("Cross-Origin-Opener-Policy", "same-origin");
-                    res.set_content(reinterpret_cast<const char*>(index_html_gz), index_html_gz_len, "text/html; charset=utf-8");
                 }
                 return false;
             });
